src/integrated/soil_canopy_model.jl

```diff
@@ -421,36 +441,76 @@ function soil_boundary_fluxes!(
         -p.soil.R_n + p.soil.turbulent_fluxes.lhf + p.soil.turbulent_fluxes.shf
 end
 
-
 """
-     PrognosticSoil{FT} <: AbstractSoilDriver
+     PrognosticSoilConditions{F <: Union{AbstractFloat, ClimaCore.Fields.Field}} <: Canopy.AbstractGroundConditions
 
-Concrete type of AbstractSoilDriver used for dispatch in cases where both
-a canopy model and soil model are run.
-$(DocStringExtensions.FIELDS)
+A type of Canopy.AbstractGroundConditions to use when the soil model is prognostic and
+of type `EnergyHydrology`. This is required because the canopy model needs albedo of the ground
+in order to compute its update_aux! function, and that function must only depend on the canopy model.
+
+In the future, we will allocate space for albedo in the cache. In that case, we would *not*
+store them here, twice. `PrognosticSoilConditions` would
+then just be a flag, essentially.
```

Comment:
this is a good question. RIght now I am kind of stuck because our "ground_albedo" functions are used in the canopy update_aux! function, which, because of our "standalone" design for models, can only access things stored in `canopy`. In the prescribed case this causes no issues, because we have the `ground` object that stores the albedo. But for the prognostic ground case, Im not sure how to dispatch off of the soil model (or soil + snow models..) since they arent in `canopy`. so the solution was to make a new`ground` with `typeof(ground) <: PrognosticSoilConditions` that instructs the canopy model where to look for the albedo (currently in `ground`, later in `p`.)

A separate issue is that in the future the albedo will be time varying/depend on soil and snow aux vars. so I actually think we may need to move some of the functions in canopy `update_aux` to later in the tendency function, since they depend on things that only get updated in the soil/snow `update_aux` function, and we dont really want to assume an ordering between these.

Reply:
I sort of think if we solve the latter, it will help us with the former. because the land model `update_boundary_fluxes!` function can handle some of the interactions (e.g. update p.alpha_ground, and *then* call radiative transfer, and then update boundary fluxes of the component models).

Reply:
that makes sense to me! I think the current solution is good given the constraints we have, and we can revisit it down the line if/when things change